src/main/java/frc/robot/vision/sources/LimeLightSource.java

```diff
@@ -0,0 +1,153 @@
+	private boolean useOldRobotPoseEntry;
+
+	public LimeLightSource(String name, String parentLogPath, Filter<RawVisionAprilTagData> filter) {
+		super(parentLogPath + name + "/");
+
+		this.name = name;
+		this.filter = filter;
+		this.robotPoseEntry = getLimelightNetworkTableEntry("botpose_orb_wpiblue");
+		this.oldRobotPoseEntry = getLimelightNetworkTableEntry("botpose_wpiblue");
+		this.aprilTagPoseEntry = getLimelightNetworkTableEntry("targetpose_cameraspace");
+		this.aprilTagIdEntry = getLimelightNetworkTableEntry("tid");
+		this.robotOrientationEntry = getLimelightNetworkTableEntry("robot_orientation_set");
+		this.gyroAngleValues = new GyroAngleValues(Rotation2d.fromDegrees(0), 0, Rotation2d.fromDegrees(0), 0, Rotation2d.fromDegrees(0), 0);
+		this.useOldRobotPoseEntry = false;
+
+		AlertManager.addAlert(
+			new PeriodicAlert(
+				Alert.AlertType.WARNING,
+				this.getLogPath() + "DisconnectedAt",
+				() -> getLimelightNetworkTableEntry("tv").getInteger(-1) == -1
+			)
+		);
+	}
+
+	public void switchToOldBotPose(boolean useOldRobotPose) {
+		useOldRobotPoseEntry = useOldRobotPose;
+	}
+
+	@Override
+	public void updateGyroAngles(GyroAngleValues gyroAngleValues) {
+		this.gyroAngleValues = gyroAngleValues;
+	}
+
+	@Override
+	public void updateEstimation() {
+		robotOrientationEntry.setDoubleArray(
+			new double[] {
+				gyroAngleValues.yaw().getDegrees(),
+				gyroAngleValues.yawRate(),
+				gyroAngleValues.pitch().getDegrees(),
+				gyroAngleValues.pitchRate(),
+				gyroAngleValues.roll().getDegrees(),
+				gyroAngleValues.rollRate()}
+		);
+		robotPoseArray = (useOldRobotPoseEntry ? oldRobotPoseEntry : robotPoseEntry)
+			.getDoubleArray(new double[VisionConstants.LIMELIGHT_ENTRY_ARRAY_LENGTH]);
+		aprilTagPoseArray = aprilTagPoseEntry.getDoubleArray(new double[VisionConstants.LIMELIGHT_ENTRY_ARRAY_LENGTH]);
+		double[] robotPoseWithoutGyroInput = oldRobotPoseEntry.getDoubleArray(new double[VisionConstants.LIMELIGHT_ENTRY_ARRAY_LENGTH]);
+		robotHeading = Rotation2d.fromDegrees(robotPoseWithoutGyroInput[LimelightEntryValue.YAW_ANGLE.getIndex()]);
+	}
+
+	private Optional<Pair<Pose3d, Double>> getUpdatedPose3DEstimation() {
+		int id = (int) aprilTagIdEntry.getInteger(VisionConstants.NO_APRILTAG_ID);
+		if (id == VisionConstants.NO_APRILTAG_ID) {
+			return Optional.empty();
+		}
+
+		double processingLatencySeconds = Conversions.milliSecondsToSeconds(robotPoseArray[LimelightEntryValue.TOTAL_LATENCY.getIndex()]);
+		double timestamp = TimeUtils.getCurrentTimeSeconds() - processingLatencySeconds;
+
+		Pose3d robotPose = new Pose3d(
+			getPoseValue(LimelightEntryValue.X_AXIS),
+			getPoseValue(LimelightEntryValue.Y_AXIS),
+			getPoseValue(LimelightEntryValue.Z_AXIS),
+			new Rotation3d(
+				Math.toRadians(getPoseValue(LimelightEntryValue.ROLL_ANGLE)),
+				Math.toRadians(getPoseValue(LimelightEntryValue.PITCH_ANGLE)),
+				Math.toRadians(getPoseValue(LimelightEntryValue.YAW_ANGLE))
+			)
+		);
```

Comment:
The LimelightSources' estimated pose should be of botpos1 or 2s' pose and the heading should be of botpos1

Reply:
I don't agree. Botpose1 and 2 gives to separate positions, and I wouldn't like to combine them together without it being clear.

Reply:
@YHerm 
@NoamRosenberg08